src/controller/controller.ts

```diff
@@ -0,0 +1,138 @@
+import { Request, Response, NextFunction } from "express";
+import { genSaltSync, hashSync, compareSync } from "bcryptjs";
+import UserModel from "../models/users";
+import bcrypt from "bcrypt";
+import jwt from "jsonwebtoken";
+
+interface UserInterface {

```

Comment:
we can create a class and implement all necessary method. also the controller.ts should be reame to userController.ts